File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
          int overAllsum=nums[0];
      int currentSum=0;
        for (int i = 0; i < nums.length; i++) {
             currentSum=Math.max(nums[i],nums[i]+currentSum);
             overAllsum=Math.max(currentSum,overAllsum);

        }
        return overAllsum;
    }
}

File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let res=nums[0]
    let sum=0;
    for(let a of nums){
        if(sum <0){
            sum=0;
        }
        sum+=a;
        res=Math.max(res,sum);
    }
    return res;
   
};